pkg/basetypes.go

```diff
@@ -154,3 +154,25 @@ type ReportedUpkeep struct {
 }
 
 type BlockHistory []BlockKey
+
+type UpkeepState uint8
+
+const Performed UpkeepState = iota
+
+type UpkeepStateReader interface {
+	// SelectByID retrieves a single upkeep state
+	SelectByID(ID string) (*UpkeepPayload, UpkeepState, error)
+	// SelectByBlock retrieves upkeep states at a specific block
+	SelectByBlock(block int64) ([]*UpkeepPayload, []UpkeepState, error)
+	// SelectByBlockRange retrieves upkeep states within block range from start (inclusive) to end (exclusive)
+	SelectByBlockRange(start, end int64) ([]*UpkeepPayload, []UpkeepState, error)
+	// SelectByUpkeepID retrieves upkeep states for an upkeep
+	SelectByUpkeepID(upkeepId *big.Int) ([]*UpkeepPayload, []UpkeepState, error)
+	// SelectByUpkeepIDs retrieves upkeep states for provided upkeeps
+	SelectByUpkeepIDs([]*big.Int) ([]*UpkeepPayload, []UpkeepState, error)
+}
```

Comment:
I don't think we need this interface here. It isn't used by the plugin so should probably be put in the core node instead.

Reply:
done